Source/ablastr/utils/Communication.cpp

```diff
@@ -60,27 +62,18 @@ void ParallelAdd(amrex::MultiFab &dst, const amrex::MultiFab &src, int src_comp,
                                                 do_single_precision_comms, period, amrex::FabArrayBase::ADD);
 }
 
-void FillBoundary (amrex::MultiFab &mf, bool do_single_precision_comms, const amrex::Periodicity &period)
+void FillBoundary (amrex::MultiFab &mf,
+                   amrex::IntVect ng,
+                   bool do_single_precision_comms,
+                   const amrex::Periodicity &period,
+                   std::optional<bool> nodal_sync)
 {
     BL_PROFILE("ablastr::utils::communication::FillBoundary");
 
-    if (do_single_precision_comms)
-    {
-        mf.FillBoundary<comm_float_type>(period);
-    }
-    else
-    {
-        mf.FillBoundary(period);
-    }
-}
-
-void FillBoundary(amrex::MultiFab &mf,
-                  amrex::IntVect ng,
-                  bool do_single_precision_comms,
-                  const amrex::Periodicity &period,
-                  const bool nodal_sync)
-{
-    BL_PROFILE("ablastr::utils::communication::FillBoundary");
+    // allow developers to always enforce nodal sync
+    bool do_nodal_sync = nodal_sync.value_or(false);
+    amrex::ParmParse pp_ablastr("ablastr");
+    pp_ablastr.query("fillboundary_always_sync", do_nodal_sync);
```

Comment:
Update me: this needs to be fixes so that `fillboundary_always_sync = 0` does not become an "always no-sync"

Reply:
Thx @RemiLehe